src/test/java/com/flightstats/hub/util/IntegrationUdpServer.java

```diff
@@ -1,74 +1,71 @@
+    private ExecutorService executorService;
 
     private final Map<String, String> store = new HashMap<>();
 
-    private final CompletableFuture<Map<String, String>> serverFuture = CompletableFuture.supplyAsync(() -> {
-        try {
-            DatagramSocket serverSocket = new DatagramSocket(port);
-            while (listening) {
-                byte[] data = new byte[70];
-                String result = openSocket(data, serverSocket);
-                Thread.sleep(500);
-                addValueToStore(result);
-            }
-        } catch(Exception ex) {
-            listening = false;
-        }
-//        logger.info(":::::::::, {}", store);
-        return store;
-    });
+    public CompletableFuture<Map<String, String>> getServerFuture() {
+        return  CompletableFuture.supplyAsync(() -> {
+            try {
+                DatagramSocket serverSocket = new DatagramSocket(port);
+                while (listening) {
+                    byte[] data = new byte[70];
+                    DatagramPacket receivePacket = new DatagramPacket(data, data.length);
+                    startupCountDownLatch.countDown();
+                    serverSocket.receive(receivePacket);
 
-    private String openSocket(byte[] data, DatagramSocket serverSocket) {
-        DatagramPacket receivePacket = new DatagramPacket(data, data.length);
-        try {
-            serverSocket.receive(receivePacket);
-            InputStream inputStream = new ByteArrayInputStream(receivePacket.getData());
-            InputStreamReader streamReader =  new InputStreamReader(inputStream, StandardCharsets.UTF_8);
-            BufferedReader reader = new BufferedReader(streamReader);
-            String result = reader
-                    .lines()
-                    .collect(Collectors.toList())
-                    .get(0)
-                    .trim();
-            if (result.contains("closeSocket")) {
+                    String result = listen(receivePacket);
+                    addValueToStore(result);
+                }
+            } catch (IOException e) {
                 listening = false;
             }
-            reader.close();
-            streamReader.close();
-            return result;
-        } catch (IOException ex) {
-            logger.error("error io exception at open socket", ex);
-            serverFuture.completeExceptionally(ex);
-            return "";
+//        log.info(":::::::::, {}", store);
+            return store;
+        }, executorService);
+    }
+
+    private String listen(DatagramPacket receivePacket) throws IOException {
+        InputStream inputStream = new ByteArrayInputStream(receivePacket.getData());
+        InputStreamReader streamReader =  new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+        BufferedReader reader = new BufferedReader(streamReader);
+        String result = reader
+                .lines()
+                .collect(Collectors.toList())
+                .get(0)
```

Comment:
Can you use Stream.findFirst()? No point in reading all the lines() if you just need the first one. Unless a full read is required to clear the socket?